sunpy/util/util.py

```diff
@@ -212,3 +212,38 @@ def expand_list_generator(inp):
                 yield nested_item
         else:
             yield item
+
+def partial_match(key, dictionary):
+    """
+    Return value/values based on the partial key.
+    Each element of the partial key is matched against the keys of the dictionary and if a match is found 
+    the value of the key is returned.
+    Even a partial match works here i.e even if the key matches partially a value is returned.
+    Note that this function uses a generator so we need to type cast the output to a list to get the values.
+
+    Parameters
+    ----------
+    key : `tuple`
+          A tuple containing the partial key.
+    dictionary: `dict`
+          The target dictionary from which we want to retrieve the value based on the partial key.
```

Comment:
To make it more general, could this second argument be just a list of tuples? I'm thinking that way could be used in the case of dictionaries like:
```python
partial_match(('a','b', None), d.keys())
```
Or maybe... if we want this to be dictionaries, then `partial_key_match` makes more sense for the function name. So with the `key` in the name I can think about it as the `key` of a dictionary.
But... since it return the values... It's not a matching function, but a `getting` one.

I don't know... too many questions...

Reply:
I have to check if the 2nd argument can be a list of tuples. 
I will change the name if I am not able to match it in tuples

Reply:
Change the function is easy, as you only need to change the `dictionary.keys()` for that input. But, if the function is returning the values of these dictionaries, then yeah, update the function name.

Reply:
Yes it is returning the values of the dictionary so I guess I will change the function name. It will be a general function to fetch values from a dictionary given a partial key